src/Pact/Eval.hs

```diff
@@ -1173,7 +1173,16 @@ functionApp fnName funTy mod_ as fnBody docs ai = do
   let body = instantiate (resolveArg ai args') fnBody
       fname = asString fnName
       fa = FunApp ai fname mod_ Defun (funTypes fty) docs
-  guardRecursion fname mod_ $ appCall fa ai args' $ fmap (gas,) $ reduceBody body
+
+  returnVal :: Term Name <-
+    guardRecursion fname mod_ $ appCall fa ai args' $ fmap (gas,) $
+    reduceBody body
+
+  whenExecutionFlagSet FlagDisablePact46 $
```

Comment:
Shouldn't this be something like `unlessExecutionFlagSet FlagDisablePact47`? It seems odd that I would want this 4.7 functionality when I'm disabling 4.6 functionality...

Reply:
I'm not quite clear actually on how we say "This code should run at 4.7 and higher".

Reply:
@jwiegley this is idiomatic for flags, new features come in "unless the latest version is disabled".

Reply:
this should be `DisablePact47`

Reply:
Ok, I see how the flags are meant to be read now and updated the PR. `DisablePact47` is the only flag used.

Type checking happens unless DisablePact47 is set.

The tests first use normal behavior (no flag set, typechecking turned on), then they use the prior behavior (DisablePact47 is set, typechecking turned off).